src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/TestResources/EndpointResource.cs

```diff
@@ -32,10 +32,22 @@ public object Put()
                     {
                         host = new ServiceHost(typeof(ServiceType));
                         // URI assumes localhost but that instead should be passed in
+                        // For Kerberos, localhost cannot be used as localhost will fall back to NTLM
+                        // this is a workaround until the URI is passed in
+                        Binding binding = GetBinding();
+                        string baseAddress = "localhost";
+                        if (binding is BasicHttpBinding)
```

Comment:
I don't think this logic belongs here in the base class.  We need a new subclass of EndpointResource (HttpWindowsResource?) that knows to use the MachineName.  If we need to add to the EndpointResource API to provide a virtual method to build the URI, then I'd go ahead and do that at the same time.